mixins/introspection.mixin.tf

```diff
@@ -0,0 +1,34 @@
+# <-- BEGIN DOC -->
+#
+# This mixin is meant to be added to Terraform components in order to append a `Component` tag to all resources in the
+# configuration, specifying which component the resources belong to.
+#
+# It's important to note that all modules and resources within the component then need to use `module.introspection.context`
+# and `module.introspection.tags`, respectively, rather than `module.this.context` and `module.this.tags`.
+#
+# <-- END DOC -->
+locals {
+  # Throw an error if lookup fails
+  check_required_tags = [
+    for k in var.required_tags :
+    lookup(module.this.tags, k)
+  ]
```

Comment:
```suggestion
  check_required_tags = module.this.enabled ? [
    for k in var.required_tags :
    lookup(module.this.tags, k)
  ] : []
```

Reply:
@nitrocode  let's do `local.enabled` to follow our pattern

Reply:
the issue with that is that if `local.enabled` is already set in `main.tf` then it would conflict with `introspection.mixin.tf` and if it’s not in the `main.tf`, then `introspection.mixin.tf` would assume that it’s there which would be more work for the user so that’s why originally `module.this.enabled` is used instead of `local.enabled`